app/main.py

```diff
@@ -1,8 +1,26 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list[dict]) -> list[Person]:
+    person_list = []
+
+    for person in people:
+        instance = Person(person["name"], person["age"])
+
+        if "wife" in person and person["wife"]:
+            instance.wife = Person.people.get(person["wife"])
+            if instance.wife:
+                instance.wife.husband = instance
+        if "husband" in person and person["husband"]:
+            instance.husband = Person.people.get(person["husband"])
+            if instance.husband:
+                instance.husband.wife = instance
+        person_list.append(instance)
+
+    return person_list
```

Comment:
This is much better, but consider also this approach:
```suggestion
def create_person_instance(person_data: dict) -> Person:
    return Person(person_data["name"], person_data["age"])


def link_spouses(person_instance: Person, person_data: dict):
    spouse_key = "wife" if "wife" in person_data else "husband"
    spouse_name = person_data.get(spouse_key)

    if spouse_name:
        spouse_instance = Person.people.get(spouse_name)
        if spouse_instance:
            setattr(person_instance, spouse_key, spouse_instance)
            setattr(spouse_instance, "husband" if spouse_key == "wife" else "wife", person_instance)


def create_person_list(people: list[dict]) -> list[Person]:
    """Create a list of Person instances from the provided list of people data."""
    person_list = []

    for person_data in people:
        person_instance = create_person_instance(person_data)
        link_spouses(person_instance, person_data)
        person_list.append(person_instance)

    return person_list
```